scripts/quests/crystalWar/WOTG_BAS_0_The_Fighting_Fourth.lua

```diff
@@ -0,0 +1,265 @@
+-----------------------------------
+-- The Fighting Fourth (WOTG Nation Quests Bastok Intro)
+-- !addquest 7 7
+-- Turbulent Storm : !pos 422.461 -48.000 -47.308 175
+-- Adelbrecht      : !pos -325.766 -12.601 -76.977 87
+-- Gebhardt        : !pos 206.299 -20.826 669.125 88
+-- Roderich        : !pos -400.039 39.991 -90.445 88
+-- Barricade       : !pos -514.960 37.979 583.287 88
+-----------------------------------
+local marketsID = zones[xi.zone.BASTOK_MARKETS_S]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.CRYSTAL_WAR, xi.quest.id.crystalWar.THE_FIGHTING_FOURTH)
+
+quest.reward =
+{
+    -- TODO: The messaging for this should be first keyItem and then item, but is swapped
+    item    = xi.item.SPRINTERS_SHOES,
+    keyItem = xi.keyItem.BRONZE_RIBBON_OF_SERVICE,
+    title   = xi.title.FOURTH_DIVISION_SOLDIER,
+
+    -- TODO: You should only get the item reward the first time you sign up to a campaign allegiance.
+    --     : You shouldn't get it again if you switch allegiances (but not a huge deal, these are a cheap item)
+}
+
+local removeRations = function(player)
+    -- Note the messaging specific to this interaction!
+    player:delKeyItem(xi.keyItem.BATTLE_RATIONS)
+    player:messageSpecial(marketsID.text.ARE_TAKEN_AWAY_FROM_CHAR, xi.keyItem.BATTLE_RATIONS)
+end
+
+local returnLetter = function(player)
+    -- Note the messaging specific to this interaction!
+    npcUtil.giveKeyItem(player, xi.keyItem.BLUE_RECOMMENDATION_LETTER, marketsID.text.ITEM_RETURNED_TO_CHAR)
+end
+
+local quitQuest = function(player)
+    player:delQuest(xi.questLog.CRYSTAL_WAR, xi.quest.id.crystalWar.THE_FIGHTING_FOURTH)
+end
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE
+            -- TODO: Conditions to not have a different campaign allegiance?
+        end,
+
+        [xi.zone.THE_ELDIEME_NECROPOLIS_S] =
+        {
+            ['Turbulent_Storm'] =
+            {
+                onTrigger = function(player, npc)
+                    -- TODO: Conditions to not have a different campaign allegiance?
+
+                    local hasLetter  = player:hasKeyItem(xi.keyItem.BLUE_RECOMMENDATION_LETTER)
+                    local hasRations = player:hasKeyItem(xi.keyItem.BATTLE_RATIONS)
+
+                    -- TODO: Verify the check for rations exists on retail
+                    if hasLetter or hasRations then
+                        -- You won't be considered a member of the Fighting Fourth until you take that blue recommendation letter to Bastok...
+                        return quest:progressEvent(8)
```

Comment:
It's my understanding that if the event doesn't progress the quest in some way, it should simply call:
```suggestion
                        return quest:event(8)
```